MaxText/integration/tunix/tunix_adaptor.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+from typing import Optional, Tuple, Any
+
+import jax
+import jax.numpy as jnp
+from flax import nnx
+
+from MaxText.common_types import MODEL_MODE_TRAIN, MODEL_MODE_AUTOREGRESSIVE, Config
+from MaxText.layers.models import Transformer
+
+Array = jax.Array  # Alias for clarity
+
+class TunixMaxTextLlama(nnx.Module):
+    """Adapter exposing Tunix SFT call signature over a LlamaTransformerNNX model."""
+
+    def __init__(
+        self,
+        base_model: Transformer,
+        *,
+        use_attention_mask: bool = False,
+    ):
+        super().__init__()
+        self.base = base_model
+        self.use_attention_mask = use_attention_mask
+
+    # ------------------------------------------------------------------ #
+    # Tunix call signature (SFT)
+    # ------------------------------------------------------------------ #
+    def __call__(
+        self,
+        input_tokens: Array,             # [B, L]
+        positions: Array,                # [B, L]
+        cache: Optional[Any],            # Tunix passes None in SFT
+        attention_mask: Optional[Array], # [B, L, L] or None
```

Comment:
how does it integrate with Flash attention and lazy attention masking?

Reply:
discussed offline. MaxText creates own attention mask